### PR TITLE
Added code to read tokens.json and sort the tokens into defi, goverance, assets, collectibles

### LINKED ISSUE #3581: Read tokens.json and put tokens into the correct group in Wallet tab
Part of #3546
Depends on: #3580

A token matches if the contract and chain ID matches an entry in each array.

This file: https://github.com/AlphaWallet/tokens/blob/main/tokens.json (check update: https://github.com/AlphaWallet/alpha-wallet-ios/issues/3581#issuecomment-1019835025)

"All" should contain every token that is not hidden (like it is before the PR).

If a token is not categorized, display it under "Assets".

Just ignore the tokens tagged as "Collectibles" (if there are any) in the JSON. We still show all and only ERC721 and ERC1155 in the hardcoded Collectibles tab. Otherwise it wouldn't work.

### PR DESCRIPTION
**Pull request by eviltofu**
Closes #3581
Simulator Screen Shot - iPhone 11 - 2022-04-12 at 19 17 52 copy
Old

Simulator Screen Shot - iPhone 11 - 2022-04-12 at 19 16 51 copy
New

### CODE REVIEW

**Comment by eviltofu** [comment]
I don't know if there tokens.json file needs to be updated. I can't tell if the defi / governance tabs are properly working.

**Review by hboon — CHANGES_REQUESTED** [review]
> I don't know if the tokens.json file needs to be updated.

It's the correct version.

> I can't tell if the defi / governance tabs are properly working. From the images above, you can see that the assets tabs are both different.

After the changes, it's likely that DeFi and Governance that will be empty or have only a few tokens. You can manually add a toke you know belong to those 2 tabs by:

1. Wallet tab
2. Tap …
3. Add/Hide Tokens
4. Paste and follow through to add (on the right chain)

**Comment by eviltofu** [comment]
I'm looking at a coinbase [article](https://www.coindesk.com/learn/what-are-defi-tokens/) and it says Link, Dai and Luna are DeFi but I've added those and the DeFi tab is still blank. @hboon

**Comment by eviltofu** [comment]
> > I don't know if the tokens.json file needs to be updated.
> 
> It's the correct version.
> 

I've checked the git log. It's not been modified since I added it?

**Comment by hboon** [comment]
Will check when I’m back but check against tokens.json that you embed, not against an external article. If you check against the article, update tokens.json instead.

**Comment by hboon** [comment]
> I've checked the git log. It's not been modified since I added it?

No, the copy you used hasn't been committed yet, so it's the lastest we have.

**Comment by hboon** [comment]
> Will check when I’m back but check against tokens.json that you embed, not against an external article. If you check against the article, update tokens.json instead.

@eviltofu So if you add the contract `0x21bfbda47a0b4b5b1248c767ee49f7caa9b23697` (chaindId = 1), the first entry in `tokens.json` you'll see it appear under the Governance tab. This test will only verify if the tokens.json is parsing correctly of course. Unless you are referring to testing something else?

File: AlphaWallet.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
-    "pins": [
-      {
-        "package": "WalletConnect",
-        "repositoryURL": "https://github.com/WalletConnect/WalletConnectSwiftV2",
-        "state": {
-          "branch": null,
-          "revision": "5ea0265d34f18ec48d71fc299adacecdacc904bb",
-          "version": "0.5.0"
-        }
+  "pins" : [
```
Comment (hboon) [inline AlphaWallet.xcworkspace/xcshareddata/swiftpm/Package.resolved:2]: Can you rebase master (or one of the earlier ones which still builds like `31d3fe25affce53c9ce42464b653b9bd97b87537`)? It'll get rid of this in the PR.

**Review by hboon — APPROVED** [review]
Can merge after rebasing.

**Comment by hboon** [comment]
Closes #3546

**Pull request merged**